### PR TITLE
Add sync committee tests with exited and withdrawable members

### PR DESCRIPTION
**Pull request by ralexstokes**
It is possible for members of the current sync committee to be in the exited or withdrawable validator lifecycle states.

This PR adds (spec) tests to ensure clients can handle these cases.

### CODE REVIEW

File: tests/core/pyspec/eth2spec/test/altair/block_processing/sync_aggregate/test_process_sync_aggregate.py [hunk 1]
```diff
@@ -402,3 +407,189 @@ def test_proposer_in_committee_with_participation(spec, state):
...
+        next_epoch_via_block(spec, state)
+
+    committee_indices = compute_committee_indices(spec, state)
+    rng = random.Random(1010)
+
+    exited_validator_index = rng.sample(committee_indices, 1)[0]
+    exits = prepare_signed_exits(spec, state, [exited_validator_index])
+    assert len(exits) == 1
+    voluntary_exit = exits.pop()
+    spec.process_voluntary_exit(state, voluntary_exit)
```
Comment (djrtwo) [inline tests/core/pyspec/eth2spec/test/altair/block_processing/sync_aggregate/test_process_sync_aggregate.py:567]: Why do the `process_voluntary_exit` and full 4-epoch transition to get to the exit vs just setting exit_epoch and withdrawable_epoch?

Comment (djrtwo) [reply tests/core/pyspec/eth2spec/test/altair/block_processing/sync_aggregate/test_process_sync_aggregate.py:567]: I understand that it leads to a more likely correct situation but I'm not certain the overhead is worth it here, especially when you artificially reduce the `withdrawable_epoch` anyway

**Review by djrtwo — COMMENTED** [review]
Test looks good. I recommend reducing the copy paste with a runner function. Or at least a function in the middle that  can be called to prepare an exited and/or withdrawable validator

You also *could* remove a lot of the logic by manually manipulating exit_epoch and withdrawable_epoch. that said, your call

File: tests/core/pyspec/eth2spec/test/altair/block_processing/sync_aggregate/test_process_sync_aggregate.py (hunk 1 above)
Comment (ralexstokes) [reply tests/core/pyspec/eth2spec/test/altair/block_processing/sync_aggregate/test_process_sync_aggregate.py:567]: that's true, was aiming for whatever is more "realistic" but can just manually set the exit and withdrawable epochs here

Comment (ralexstokes) [reply tests/core/pyspec/eth2spec/test/altair/block_processing/sync_aggregate/test_process_sync_aggregate.py]: refactored some of the duplicated code and moved to manually setting exit and withdrawable epochs

**Review by djrtwo — COMMENTED** [review]
very nice

**Review by djrtwo — APPROVED** [review]
very nice (this time clicked approve)!

**Pull request merged**